documentation/library_guide/kernel_templates_main.rst

```diff
@@ -0,0 +1,17 @@
+Kernel Templates API
+####################
+
+Kernel Templates is an experimental API that includes algorithms which can be configured based on a variety of parameters, such as the number of elements to be processed by a work-item and the size of a work-group.
```

Comment:
My suggestions: 
Kernel Templates is an experimental API that includes algorithms that can be configured based on various parameters, such as the number of elements to be processed by a work item and the size of a work group. Use it when you can specialize an algorithm for a specific workload (for example, the number of elements and their type) or a device to achieve better performance.